src/drivers/boards/bbblue/pwr.cpp

```diff
@@ -0,0 +1,71 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2018 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/**
+ * @file pwr.cpp
+ *
+ * System power related functions
+ *
+ * @author: UAV-Pilot
+ */
+
+/************************************************************************************
+ * Included Files
+ ************************************************************************************/
+#include <unistd.h>
+#include <linux/reboot.h>
+#include <sys/reboot.h>
+
+#include <stddef.h>
+#include <px4_log.h>
+#include <px4_config.h>
+
+#include "board_config.h"
+
+int board_register_power_state_notification_cb(power_button_state_notification_t cb) { return 0; }
+
+int board_shutdown(void)
+{
+	PX4_INFO("Powering off board ...");
+
+	// root is required by librobotcontrol
+	// so with BeagleBone Blue, it's running with sufficient priviledges
+	sync();
+	reboot(LINUX_REBOOT_CMD_POWER_OFF);
```

Comment:
I don't see why this is should be beaglebone blue specific. Let's try to keep things generic for broader linux usage if possible.

Reply:
deleted pwr.cpp and remove the code to a common file